controllers/aggregated_clusteroperator_controller.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	utilerror "k8s.io/apimachinery/pkg/util/errors"
+
+	openshiftconfigv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"k8s.io/client-go/discovery"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logr "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	platformv1alpha1 "github.com/openshift/platform-operators/api/v1alpha1"
+	aggregatedco "github.com/openshift/platform-operators/internal/aggregated-co"
+	"github.com/openshift/platform-operators/internal/util"
+)
+
+type AggregatedClusterOperatorReconciler struct {
+	client.Client
+	DiscoveryClient discovery.DiscoveryInterface
+	Configv1Client  configv1client.ConfigV1Interface
+}
+
+const aggregateCOName = "platform-operators-aggregated"
+
+// TODO kubebuilder markers
+
+func (a *AggregatedClusterOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logr.FromContext(ctx)
+	log.Info("reconciling request", "req", req.NamespacedName)
+	defer log.Info("finished reconciling request", "req", req.NamespacedName)
+
+	if req.Name != aggregateCOName {
+		return ctrl.Result{}, nil
+	}
+
+	// Create a builder to update the CO status
+	builder := aggregatedco.NewBuilder()
+	// Create a writer to write to the CO status
+	writer := aggregatedco.NewWriter(a.DiscoveryClient, a.Configv1Client)
+
+	aggregatedCO := &openshiftconfigv1.ClusterOperator{}
+	if err := a.Get(ctx, req.NamespacedName, aggregatedCO); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	defer func() {
+		if err := writer.UpdateStatus(aggregatedCO, builder.GetStatus()); err != nil {
+			log.Error(err, "error updating CO status")
+		}
+	}()
+
+	// Set the default CO status conditions: Progressing True, Degraded False, Available False
+	builder.WithProgressing(openshiftconfigv1.ConditionTrue, "")
+	builder.WithDegraded(openshiftconfigv1.ConditionFalse)
+	builder.WithAvailable(openshiftconfigv1.ConditionFalse, "", "")
+
+	POList := &platformv1alpha1.PlatformOperatorList{}
```

Comment:
nit:  don't capitalize non-global variable names.
```suggestion
	poList := &platformv1alpha1.PlatformOperatorList{}
```